windows/gce-gpu-monitoring-cuda.ps1

```diff
@@ -0,0 +1,227 @@
+    #>
+    
+    $query_params = 'gpu_name,gpu_bus_id,'+(($gpu_metrics.Keys|sort-object) -join ",")
+
+    $result = & $nvidia_smi_path --query-gpu=$query_params --format=csv,nounits,noheader
+
+    $metric_name = $query_params -split ","
+
+    $final = @{}
+    foreach($r in $result -split "\r?\n"){
+        $r = ($r -split ",").trim()
+        for( $i =2;$i -lt $r.length;$i++){
+            $final[$r[0]+","+$r[1]]+=@{$metric_name[$i]=$r[$i]}
+        }
+    }
+
+    return $final
+    
+}
+
+function get_timeseries_entry{
+    param(
+        $metric_time, 
+        $nvidia_metric,
+        $gcp_metric_name,
+        $value,
+        $gpu_bus_id,
+        $gpu_type
+        )
+
+
+   @{'metric'=@{
+        'type'="custom.googleapis.com/$gcp_metric_name"
+        'labels'=@{
+            'gpu_type'= $gpu_type
+            'gpu_bus_id'= $gpu_bus_id
+        }
+        }
+        
+    'resource'=@{
+        'type'='gce_instance'
+        'labels'=@{
+            'project_id'= $project_id
+            'instance_id'=$instance_id
+            'zone'=$zone
+            }
+        }
+
+    'points' = @(
+            @{
+                'interval'=@{
+                    'endTime'=$metric_time
+                    }
+                'value'=@{
+                    'int64Value'= $value
+                    }
+                    
+            }        
+        )
+    }
+}
+
+function check_nvidia_smi{
+    <#
+    Checks if the nvidia-smi tool is installed and if it detects any GPUs.
+    Prints message to stderr in case of errors.
+    =return=
+    True if the nvidia-smi tool is available.
+    #>
+
+    $installed = test-path -Path $nvidia_smi_path
+    if (-not($installed)){
+        write-host -ForegroundColor Red "Couldn't find the nvidia-smi tool. Make sure it's properly installed in one of the directories in $nvidia_smi_path."
+        return $false
+    }
+
+    $result = & $nvidia_smi_path -L 
+    if ($LASTEXITCODE -ne 0){
+        write-host -ForegroundColor Red "The nvidia-smi tool has encountered an error= "
+        write-host -ForegroundColor Red "$result"
+        return $false
+    }
+
+    <# 
+    $no_cards = $result|Measure-Object -Line
+    if ( $no_cards -eq 0 ){
+        Write-host -ForegroundColor RED "The nvidia-smi tool didn't detect any GPUs attached to the system."
+        return $false
+    }
+    #>
+    
+    return $true
+}
+
+function report_metrics{
+    <#
+    Reports a set of metrics to the Cloud Monitoring system.
+    :param values: A dictionary mapping (gpu_type, gpu_bus_id) to a map of
+        metric names and their values (floats).
+    :param project_id: Project number of the hosting machine.
+    :param zone: Zone of the hosting machine.
+    :param instance_id: Instance ID of the hosting machine.
+    :return:
+    #>
+    param(
+        $metrics
+    )
+
+    $data = @{'timeSeries'=@()}
+
+    #gpu total
+    $now=(get-date).ToString("O")
```

Comment:
If we intend to post the metric in UTC time, it might be beneficial to ensure we are converting the timestapt to UTC, for example: `(Get-Date).ToUniversalTime().toString("O")`

Reply:
Thanks. updated.